### PR TITLE
chore: fix all eslint warnings and errors

### PR DESCRIPTION
**Pull request by SirFuzzyWalls**
## Summary

Resolves all 44 lint problems (`npm run lint` now exits clean — 0 errors, 0 warnings) and fixes a real bug uncovered during the process.

- **7 `prefer-const` errors** — auto-fixed with `eslint --fix`
- **~14 `no-unused-vars` warnings** — removed dead imports and variables across 19 files
- **9 `react-hooks/set-state-in-effect` errors** — all false positives (intentional prop-sync patterns); suppressed with targeted `eslint-disable-next-line` comments
- **1 unescaped apostrophe** — fixed in `landing-our-mission.tsx`
- **3 `@next/next/no-img-element` warnings** — replaced `` with `<Image>` in `AvatarUpload` (with Supabase domain added to `next.config` `remotePatterns`); disabled for blob URL and unknown-dimension cases with explanatory comments
- **1 `react-hooks/exhaustive-deps` warning** — fixed in `ConversationView` and `PollsCard`

## Bug fix: duplicate Realtime messages in DM

During testing, messages were appearing twice in DM conversations. Root cause: the initial exhaustive-deps fix moved `createClient()` to module scope, which caused stale Realtime channel subscriptions to persist across HMR reloads — resulting in two active subscriptions firing on every incoming message.

Fixed by using `useRef(createClient()).current` — one client instance per component mount, properly cleaned up on unmount, with a stable reference that satisfies the deps linter.

## Test plan

- [x] `npm run lint` exits with 0 problems
- [x] Send and receive DMs — messages appear once only
- [x] Avatar upload works (requires `avatars` bucket in Supabase dashboard)
- [x] `/members`, `/events`, `/groups`, `/messages`, `/profile` all render without console errors

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by SirFuzzyWalls** [comment]
Tested locally & on triggered vercel deploy without any noticeable regressions. As this was a lint fix it touched quite a few files, so maybe a lower priority review?

File: components/messages/ConversationList.tsx [hunk 1]
```diff
@@ -128,7 +130,7 @@ export function ConversationList({
           </div>
         ) : (
           conversations.map(
```
Comment (kaizengrowth) [inline components/messages/ConversationList.tsx]: Nice catch here! `groupSlug` isn't actually being used.

File: components/messages/ConversationView.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import Link from "next/link";
 import { createClient } from "@/lib/supabase/client";
 import { cn } from "@/lib/utils";
 import { UserAvatar } from "@/components/shared/UserAvatar";
-import { getAvatarColor, getInitials } from "@/lib/utils/avatar";
```
Comment (kaizengrowth) [inline components/messages/ConversationView.tsx:9]: Nice cleanup of `getInitials`

File: components/messages/ConversationView.tsx [hunk 3]
```diff
@@ -128,7 +129,7 @@ export function ConversationView({
     const interval = setInterval(fetchLastSeen, 30_000); // every 30s
     fetchLastSeen(); // run once immediately
     return () => clearInterval(interval);
-  }, [otherUser?.id]);
+  }, [supabase, otherUser?.id]);
```
Comment (kaizengrowth) [inline components/messages/ConversationView.tsx:140]: Appreciate your adding `supabase` to the dependency array here, and `useRef()` for `createClient()`. This is clean, and addresses the sloppy use of `eslint-disable`.

File: components/messages/FileAttachmentPreview.tsx [hunk 4]
```diff
@@ -35,6 +35,7 @@ export function FileAttachmentPreview({
     >
       {isImage ? (
         <div className="relative size-12 shrink-0 overflow-hidden rounded bg-muted">
+          {/* eslint-disable-next-line @next/next/no-img-element -- blob URL, next/image does not support blob: scheme */}
```
Comment (kaizengrowth) [inline components/messages/FileAttachmentPreview.tsx:38]: Great!

File: components/profile/AvatarUpload.tsx [hunk 5]
```diff
@@ -74,7 +75,7 @@ export function AvatarUpload({
           )}
         >
           {currentAvatarUrl && !uploading ? (
-            <img src={currentAvatarUrl} alt="" className="size-full object-cover" />
+            <Image src={currentAvatarUrl} alt="" width={96} height={96} className="size-full object-cover" />
```
Comment (kaizengrowth) [inline components/profile/AvatarUpload.tsx:78]: Thanks for changing this to Next Image

File: next.config.ts [hunk 6]
```diff
@@ -1,7 +1,14 @@
 import type { NextConfig } from "next";
 
 const nextConfig: NextConfig = {
-  /* config options here */
+  images: {
```
Comment (kaizengrowth) [inline next.config.ts:4]: This is great! Thanks for adding this.

File: components/messages/ConversationView.tsx [hunk 7]
```diff
@@ -82,7 +82,8 @@ export function ConversationView({
   participants = [],
   readOnlyFooter,
 }: ConversationViewProps) {
-  const supabase = createClient();
+  // Stable client instance per component mount — avoids duplicate Realtime subscriptions on HMR
+  const supabase = useRef(createClient()).current;
```
Comment (kaizengrowth) [inline components/messages/ConversationView.tsx]: Wondering if it would be better to use:

`useMemo(() => createClient(), [])`

rather than `useRef` - as it is a stable computed value.

File: components/messages/MessageBubble.tsx [hunk 8]
```diff
@@ -101,6 +98,7 @@ function FileMessageBubble({
               onClick={handleOpenUrl}
               className="block max-w-full rounded overflow-hidden focus:outline-none focus:ring-2 focus:ring-primary"
             >
+              {/* eslint-disable-next-line @next/next/no-img-element -- dimensions unknown at render time; next/image requires explicit width/height */}
```
Comment (kaizengrowth) [inline components/messages/MessageBubble.tsx:101]: next/image with `fill` mode doesn't require explicit dimensions, and sizes to its parent container.

Adding `position: relative` and a defined size to the parent `<button>` could let the rule stay enabled.

This is a good fix if the image aspect ratio is truly unpredictable, but worth a look.

File: app/(app)/profile/profile-form.tsx [hunk 9]
```diff
@@ -47,6 +47,7 @@ export default function ProfileForm({ profile }: ProfileFormProps) {
   const [loading, setLoading] = useState(false);
 
   useEffect(() => {
+    // eslint-disable-next-line react-hooks/set-state-in-effect
```
Comment (kaizengrowth) [inline app/(app)/profile/profile-form.tsx]: Suppressing this works well for now.

It may also be good to pass `key={profile.id}` to `<ProfileForm>` and remove the `useEffect` entirely, to avoid the edge case where in-progress edits might get wiped if the `profile` prop refreshes mid-edit.

File: lib/storage.test.ts [hunk 10]
```diff
@@ -1,7 +1,6 @@
 import {
   deleteFile,
   getSignedUrl,
-  uploadPrivateFile,
```
Comment (kaizengrowth) [inline lib/storage.test.ts:4]: Great to remove this. 

Also realize we need test coverage for `uploadPrivateFile`

**Review by kaizengrowth — DISMISSED** [review]

File: components/messages/ConversationView.tsx (hunk 7 above)
Comment (SirFuzzyWalls) [reply components/messages/ConversationView.tsx]: Done!

**Comment by tonyrosario** [comment]
I'm not getting any lint errors on the branch but noticed that there are 3 merge conflicts (from main) that need to be resolved.

once those are done, should be good to merge 👍

**Comment by timchaffee** [comment]
Anything left to do besides resolve merge conflicts with this?
